--- ./pypungi/pungi.py.sha1	2007-03-12 15:00:43.000000000 -0400
+++ ./pypungi/pungi.py	2007-03-12 15:01:10.000000000 -0400
@@ -176,10 +176,10 @@
                                                         isodir,
                                                         isoname,
                                                         os.path.join('%s-disc%s' % (self.topdir, disc))))
-            os.system('cd %s; sha1sum %s >> SHA1SUM' % (isodir, isoname))
             # implant md5 for mediacheck on all but source arches
             if not self.config.get('default', 'arch') == 'source':
                 os.system('/usr/lib/anaconda-runtime/implantisomd5 %s' % os.path.join(isodir, isoname))
+            os.system('cd %s; sha1sum %s >> SHA1SUM' % (isodir, isoname))
 
         # We've asked for more than one disc, and we're not srpms, so make a DVD image
         if self.config.getint('default', 'discs') > 1 and not self.config.get('default', 'arch') == 'source':
@@ -215,8 +215,8 @@
                                                         isodir,
                                                         isoname,
                                                         self.topdir))
-            os.system('cd %s; sha1sum %s >> SHA1SUM' % (isodir, isoname))
             os.system('/usr/lib/anaconda-runtime/implantisomd5 %s' % os.path.join(isodir, isoname))
+            os.system('cd %s; sha1sum %s >> SHA1SUM' % (isodir, isoname))
 
             shutil.move(os.path.join(self.config.get('default', 'destdir'), '.discinfo-%s' % self.config.get('default', 'arch')), discinfofile)
 
